Sprint 2 New

tmj1 opened the pull request.

tmj1 committed:
basic configs changes

tmj1 committed:
add file color variables

tmj1 committed:
add file color variables
refactoring code
add listMessages
add attach file function
add

tmj1 committed:
refactor code

tmj1 committed:
refactor code

tmj1 committed:
added exports utils
added ViewScreens enum
added Route class
added params for another pages
refactoring code

tmj1 committed:
added service for Auth
added register function
fixed error
refactoring code
decomposed code

tmj1 committed:
fixed style
refactoring popup component
decomposed code
fixed error
refactoring code
fixed HTTPTransport class
deleted routes

tmj1 committed:
fixed style

tmj1 committed:
refactoring code

deleted routes
refactoring code
code style
fixed error
decomposed code

tmj1 committed:
Update .nvmrc

tmj1 committed:
Update README.md

tmj1 committed:
Create .editorconfig

mikebars1995 commented on .nvmrc:
```diff
@@ -0,0 +1 @@
+
```
**Нужно исправить**


нужно ограничить версию `node` в `package.json` (или можно прописать версию в `.nvmrc`) примерно так: `"engines" : { "node" : ">=12" }`, чтобы она была не ниже `12` версии. Это есть в чек-листе

tmj1 replied:
добавлено в nvmc

mikebars1995 commented on package.json:
```diff
@@ -19,12 +35,15 @@
     "@anikin/parcel-transformer-handlebars": "^1.0.9",
-    "parcel": "^2.5.0",
-    "postcss-nested": "^5.0.6"
+    "parcel": "^2.8.0",
```
**Можно лучше**

Лучше задавать **фиксированные версии** зависимостям. Если нечаянно удалить `package-lock.json` и попытаться повторно установить зависимости с помощью команды `npm install`, то **npm** установит последние версии зависимостей, что может привести к ошибкам.

mikebars1995 commented on src/partials/avatar/avatar.ts:
```diff
@@ -0,0 +1,27 @@
+      />
+    `;
+    }
+}
```
**Нужно исправить**

В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`

tmj1 replied:
добавлены пустые строки, добавлена настройка в IDE

mikebars1995 commented on src/utils/classes/HTTPTransport.ts:
```diff
@@ -0,0 +1,62 @@
+}
+
+export class HTTPTransport {
+  public get = (url: string, options = {}) =>
```
**Можно лучше**
Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище

```jsx

// создаем тип метода
type HTTPMethod = (url: string, options?: Options) => Promise<unknown>

export class HTTPTransport {

  // используем тип и удаляем дублирование в аргументах
  get: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.GET}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  put: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.PUT}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  post: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.POST}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  delete: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.DELETE}, options.timeout)
  )

```

tmj1 replied:
спасибо за подсказку! обязательно исправлю позже, просто сейчас время поджимает очень сильно

mikebars1995 commented on src/utils/classes/HTTPTransport.ts:
```diff
@@ -0,0 +1,62 @@
+  private request = (url: string, options: RequestOptions) => {
+    const { method = METHODS.GET, headers = {}, data, timeout = 5000 } = options;
+
+    const query = method === METHODS.GET ? queryStringify(data as RequestData) : '';
```
**Можно лучше**

`queryStringify` лучше сразу в методе `get` вызывать, чтобы не проверять `METHODS.GET` при каждом запросе

tmj1 replied:
точно! спасибо!

mikebars1995 commented on src/pages/change-password/change-password.ts:
```diff
@@ -0,0 +1,115 @@
+                  formName="profile__form_el_change-password-form"
+                }}}
+                {{{Button
+                  onClick=handleSubmitForm
```
**Можно лучше**



Обработчик сабмита лучше навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`, так как сабмит формы происходит ещё при нажатии `Enter`, и он не будет работать, если навесить обработчик клика на кнопку только. **Это нужно исправить везде, где есть инпуты и форма**

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная



Что сделано отлично:

* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что фиксируете версии зависимостей
* Хорошая структура папок и файлов
* Добавлен `ESLint` и `Stylelint`
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что не забываете про тег `main` для главной части страницы



Что нужно исправить:



* Проект не запускается локально(`npm run dev`), скрин ошибки в консоли https://skr.sh/sHDkwWzkjOB
* нужно ограничить версию `node` в `package.json` (или можно прописать версию в `.nvmrc`) примерно так: `"engines" : { "node" : ">=12" }`, чтобы она была не ниже `12` версии. Это есть в чек-листе 
* В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`



Что можно улучшить:

* Лучше задавать **фиксированные версии** зависимостям. Если нечаянно удалить `package-lock.json` и попытаться повторно установить зависимости с помощью команды `npm install`, то **npm** установит последние версии зависимостей, что может привести к ошибкам.
* Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище
* Обработчик сабмита лучше навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`





Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

tmj1 committed:
Refactor code

refactoring code
- changed versions in package.json /fixed error in local start npm run dev/
- changed file nvmrc
- add settind in IDE and add line in each file

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
